Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,40 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip non-workshared documents because they can't contain worksets
+            if (document.IsWorkshared == false)
+                return false;
+
+            // Skip non-string workset names
+            string worksetName = value as string;
+            if (worksetName is string == false)
+                return false;
+
+            // Skip null or empty workset names
+            if (string.IsNullOrWhiteSpace(worksetName))
+                return false;
```

Comment:
All that code should be placed inside `if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)` - it is irrelevant if we are not trying to attempt _Workset_ parameter

Reply:
Enclosed in the if statement as suggested